frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -264,16 +263,23 @@ class FileUploader extends React.PureComponent<Props, State> {
   }
 
   private removeFile = (fileId: string): void => {
-    this.setState(state => {
-      const filteredFiles = state.files.filter(file => file.id !== fileId)
-      return {
-        status: filteredFiles.length
-          ? FileStatuses.UPLOADED
-          : FileStatuses.READY,
-        errorMessage: undefined,
-        files: filteredFiles,
-      }
-    })
+    this.setState(
+      state => {
+        const filteredFiles = state.files.filter(file => file.id !== fileId)
+        return {
+          status: filteredFiles.length
+            ? FileStatuses.UPLOADED
+            : FileStatuses.READY,
+          errorMessage: undefined,
+          files: filteredFiles,
+        }
+      },
+      () =>
+        this.props.uploadClient.updateFileCount(
+          this.props.element.id,
+          this.state.files.length
```

Comment:
quick sanity check, we don't need to add "acceptedFiles.length" in case the file uploader is in the middle of uploading?

Reply:
This callback runs after the state is updated so if there's files being uploaded, it will update the state. Passing the function in instead of an object gets the current state so it should have as close to accurate as possible. Also acceptedFiles.length would only be the number on drop so if you do 2 separate uploads, acceptedFiles.length would only be the number of files in the second upload whereas here we are updating the total count of files in the widget